backend/api/MQTT/MqttService.cs

```diff
@@ -0,0 +1,194 @@
+        public event EventHandler<MqttReceivedArgs>? MqttIsarTaskReceived;
+        public event EventHandler<MqttReceivedArgs>? MqttIsarStepReceived;
+        #endregion
+
+        public static MqttService? Instance { get; private set; }
+
+        private readonly ILogger<MqttService> _logger;
+
+        private readonly IManagedMqttClient _mqttClient;
+
+        private readonly ManagedMqttClientOptions _options;
+
+        private readonly TimeSpan _reconnectDelay = TimeSpan.FromSeconds(5);
+
+        public MqttService(ILogger<MqttService> logger, IConfiguration config)
+        {
+            Instance = this;
+            _logger = logger;
+            var mqttFactory = new MqttFactory();
+            _mqttClient = mqttFactory.CreateManagedMqttClient();
+
+            var mqttConfig = config.GetSection("Mqtt");
+
+            string username = mqttConfig.GetValue<string>("Username");
+            string password = "default";
+
+            var builder = new MqttClientOptionsBuilder()
+                .WithTcpServer(
+                    mqttConfig.GetValue<string>("Host"),
+                    mqttConfig.GetValue<int>("Port")
+                )
+                .WithCredentials(username, password);
+
+            _options = new ManagedMqttClientOptionsBuilder()
+                .WithAutoReconnectDelay(_reconnectDelay)
+                .WithClientOptions(builder.Build())
+                .Build();
+
+            RegisterCallbacks();
+
+            string[] topics = mqttConfig.GetSection("Topics").Get<string[]>();
+            foreach (string topic in topics)
+            {
+                SubscribeToTopic(topic);
+            }
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            _logger.LogInformation("MQTT client STARTED");
+            await _mqttClient.StartAsync(_options);
+            while (!stoppingToken.IsCancellationRequested) { }
+            await _mqttClient.StopAsync();
+            _logger.LogInformation("MQTT client STOPPED");
+        }
+
+        /// <summary>
+        /// The callback function for when a subscribed topic publishes a message
+        /// </summary>
+        /// <param name="messageReceivedEvent"> The event information for the MQTT message </param>
+        private void OnMessageReceived(MqttApplicationMessageReceivedEventArgs messageReceivedEvent)
+        {
+            string content = messageReceivedEvent.ApplicationMessage.ConvertPayloadToString();
+            _logger.LogInformation("Message recieved: {payload}", content);
+
+            string topic = messageReceivedEvent.ApplicationMessage.Topic;
+            var messageType = MqttTopics.TopicsToMessages.GetItemByTopic(topic);
+            if (messageType is null)
+            {
+                _logger.LogError("No message class defined for topic '{topicName}'", topic);
+                return;
+            }
+
+            switch (messageType)
```

Comment:
Can we raise an event based on the specific topic rather than grouping them all, then sorting it?

Reply:
I don't think so but i will look into this.

Reply:
No support for this in the MQTT library unfortunately